File: app/Component/Weather/Weather.py
import datetime as dt
import requests

BASE_URL = "http://api.openweathermap.org/data/2.5/weather?"
API_KEY = 'caab6267314b8e6ea84310000bb625d5'
CITY = 'Mumbai'

url = BASE_URL + "appid=" + API_KEY + "&q=" + CITY + "&units=metric"

response = requests.get(url).json()

print("Date:", dt.datetime.fromtimestamp(response['dt']))
print("Weather:", response['weather'][0]['main'])
print('Current Temperature:', response['main']['temp'], '°C')
print("Feels Like:", response['main']['feels_like'], '°C')
print("Min Temperature:", response['main']['temp_min'], '°C')
print("Max Temperature:", response['main']['temp_max'], '°C')
print('Current Humidity:', str(response['main']['humidity']) + '%')
print('Current Pressure:', response['main']['pressure'], "hPa")
print("Visibility", response['visibility'], 'm')
print("Wind Speed:", response['wind']['speed'], 'm/s')
print("Wind Direction:", str(response['wind']['deg']) + '°')
print("Clouds:", str(response['clouds']['all']) + '%')
print("Sunrise:", dt.datetime.fromtimestamp(response['sys']['sunrise']))
print("Sunset:", dt.datetime.fromtimestamp(response['sys']['sunset']))
print("Timezone:", dt.datetime.fromtimestamp(response['timezone']))

File: app/Component/Weather/Weather.js
const fetch = require('node-fetch');

const BASE_URL = "http://api.openweathermap.org/data/2.5/weather";
const API_KEY = 'caab6267314b8e6ea84310000bb625d5';
const CITY = 'Mumbai';

const url = `${BASE_URL}?q=${CITY}&units=metric&appid=${API_KEY}`;

fetch(url)
    .then(response => response.json())
    .then(data => {
        console.log("Date:", new Date(data.dt * 1000));
        console.log("Weather:", data.weather[0].main);
        console.log('Current Temperature:', data.main.temp, '°C');
        console.log("Feels Like:", data.main.feels_like, '°C');
        console.log("Min Temperature:", data.main.temp_min, '°C');
        console.log("Max Temperature:", data.main.temp_max, '°C');
        console.log('Current Humidity:', data.main.humidity + '%');
        console.log('Current Pressure:', data.main.pressure, "hPa");
        console.log("Visibility", data.visibility, 'm');
        console.log("Wind Speed:", data.wind.speed, 'm/s');
        console.log("Wind Direction:", data.wind.deg + '°');
        console.log("Clouds:", data.clouds.all + '%');
        console.log("Sunrise:", new Date(data.sys.sunrise * 1000));
        console.log("Sunset:", new Date(data.sys.sunset * 1000));
        console.log("Timezone:", new Date(data.timezone * 1000));
    })
    .catch(error => console.error("Error fetching weather data:", error));
